scripts/upower/95-upower-hid.hwdb

```diff
@@ -0,0 +1,200 @@
+##############################################################################################################
+# Uninterruptible Power Supplies with USB HID interfaces
+#
+# This file was automatically generated by NUT:
+# https://github.com/networkupstools/nut/
+#
+# To keep up to date, monitor upstream NUT
+# https://github.com/networkupstools/nut/commits/master/scripts/upower/95-upower-hid.hwdb
+# or checkout the NUT repository and call 'tools/nut-usbinfo.pl'
+
+# Hewlett Packard
+usb:v03f0p0001*
+usb:v03f0p1f06*
+usb:v03f0p1f08*
+usb:v03f0p1f09*
+usb:v03f0p1f0a*
+usb:v03f0p1fe0*
+usb:v03f0p1fe1*
+usb:v03f0p1fe2*
+usb:v03f0p1fe3*
+usb:v03f0p1fe5*
```

Comment:
For known product IDs does the line need to end with asterisk (are there further fields in the `usb:...` pattern) or can it be a faster exact match?

Reply:
Yes, and that is the usual pattern that is also used inside of systemd for example. I guess the generated binary hwdb file is optimised for prefix searches. The real string we match against is visible from `udevadm info -e` as the MODALIAS, e.g. `MODALIAS=usb:v0463pFFFFd0100dc00dsc00dp00ic03isc00ip00in00`.

Reply:
Hm, looking at that example - should "our" strings be upper-cased hex? Or does it match insensitively?

Reply:
Wow, yep, I really got that one wrong. The MODALIAS for the usb: entries in the hwdb are explicitly uppercase (while the idVendor sysfs attributes are lowercase).